conan/tools/meson/toolchain.py

```diff
@@ -336,8 +336,9 @@ def __init__(self, conanfile, backend=None, native=False):
         #: Defines the Meson ``objcpp_link_args`` variable. Defaulted to ``LDFLAGS`` build environment value
         self.objcpp_link_args = []
 
-        self._resolve_apple_flags_and_variables(build_env, compilers_by_conf)
-        if native is False:
+        if self._is_apple_system:
+            self._resolve_apple_flags_and_variables(build_env, compilers_by_conf)
+        if native is False and self.cross_build and self.cross_build["host"].get("system") == "android":
             self._resolve_android_cross_compilation()
```

Comment:
These look like a refactor. If this PR is fixing a bug it is better to not do refactors, so the diff is minimal and focus on the fix, that help reviewers a lot.